ethcore/sync/src/chain/supplier.rs

```diff
@@ -238,6 +249,10 @@ impl SyncSupplier {
 			if let Some(node) = io.chain().state_data(&r.val_at::<H256>(i)?) {
 				data.push(node);
 				added += 1;
+				// Check that the packet won't be oversized
+				if data.len() > payload_soft_limit {
```

Comment:
This seems incorrect - it's just a vector of state data objects, not bytes.

Maybe worth computing RLP inline and checking the actual size (maybe every `N` items to avoid unecessary work)? It seems that `begin_list(len)` and `begin_unbounded_list()` is exactly the same - the only difference is that you need to `complete_unbounded_list` and it will panic if you exceed declared size (TBH the bounded list API seems completely useless, CC @debris).